### ISSUE TITLE
Version 3.0.1 not working

### ISSUE THREAD

**Issue by eogienko** [serilog-contrib/serilog-sinks-graylog#95]
Code:
<img width="882" alt="image" src="https://github.com/serilog-contrib/serilog-sinks-graylog/assets/136815726/96e48997-b5bc-4001-b898-6d36f97af0a7">

appsettings.json:
<img width="455" alt="image" src="https://github.com/serilog-contrib/serilog-sinks-graylog/assets/136815726/39542e48-fe52-4cd3-a78d-dc6f54f7a467">

I had version 2.3.0 installed which worked great. Updated to 3.0.0 (3.0.1) and everything broke. Logs are not sent to Graylog.
**What to do?**

**Comment by tzographos** [comment]
Indeed, version 3.0.1 still doesn't work. Version 2.3.0 works fine.

**Comment by yyavci** [comment]
any updates? i have the same issue

**Comment by whir1** [comment]
@yyavci I can't help because, current version worked on our servers on udp  =\

**Comment by joriverm** [comment]
ive come across an issue where a call to `_logger.LogError` causes the application to freeze and .net ends up infinitely waiting for something (this is within our global exception handler, so everything is halted/frozen).

maybe this can help point towards some fix?
our settings : 
```
"Serilog": {
  "Using": [
    "Serilog.Sinks.Debug",
    "Serilog.Sinks.Graylog"
  ],
  "MinimumLevel": "Verbose",
  "WriteTo": [
    {
      "Name": "Debug"
    }
    {
      "Name": "Graylog",
      "Args": {
        "Facility": "{redacted}",
        "HostnameOrAddress": "{redacted hostname}",
        "Port": 12205,
        "TransportType": "Udp",
        "MinimumLogEventLevel": "Error",
        "UseSsl": true
      }
    }
  ]
}
```

2.3 had no issues, so somewhere between 2.3 & 3.0.2 this started happening.

**Comment by danielwertheim** [comment]
How can you not be affected by this? https://github.com/serilog-contrib/serilog-sinks-graylog/issues/97

It blows up immediatelly, during boot. Have done my own extension method that I use instead and also removed the hard coded UDP value. Then everything seem to work fine.
